cms/static/js/views/video_transcripts.js

```diff
@@ -0,0 +1,270 @@
+define(
+    ['underscore', 'gettext', 'js/views/baseview', 'common/js/components/views/feedback_prompt',
+        'edx-ui-toolkit/js/utils/html-utils', 'edx-ui-toolkit/js/utils/string-utils',
+        'text!templates/video-transcripts.underscore', 'text!templates/video-transcript-upload-status.underscore'],
+    function(_, gettext, BaseView, PromptView, HtmlUtils, StringUtils, videoTranscriptsTemplate,
+                videoTranscriptUploadStatusTemplate) {
+        'use strict';
+
+        var VideoTranscriptsView = BaseView.extend({
+            tagName: 'div',
+
+            events: {
+                'click .toggle-show-transcripts-button': 'toggleShowTranscripts',
+                'click .upload-transcript-button': 'chooseFile',
+                'click .more-details-action': 'showUploadFailureMessage'
+            },
+
+            initialize: function(options) {
+                this.transcripts = options.transcripts;
+                this.edxVideoID = options.edxVideoID;
+                this.clientVideoID = options.clientVideoID;
+                this.transcriptAvailableLanguages = options.transcriptAvailableLanguages;
+                this.videoSupportedFileFormats = options.videoSupportedFileFormats;
+                this.videoTranscriptSettings = options.videoTranscriptSettings;
+                this.template = HtmlUtils.template(videoTranscriptsTemplate);
+                this.transcriptUploadStatusTemplate = HtmlUtils.template(videoTranscriptUploadStatusTemplate);
+                this.defaultFailureTitle = gettext('Your file could not be uploaded');
+                this.defaultFailureMessage = gettext('This may be happening because of an error with our server or your internet connection. Try refreshing the page or making sure you are online.'); // eslint-disable-line max-len
+                this.transcriptUploadStatuses = {
+                    uploaded: {
+                        statusClass: 'success',
+                        iconClasses: 'fa-check',
+                        shortMessage: 'Transcript uploaded.',
+                        hiddenClass: 'hidden'
+                    },
+                    uploading: {
+                        statusClass: '',
+                        iconClasses: 'fa-spinner fa-pulse',
+                        shortMessage: 'Uploading transcript',
+                        hiddenClass: 'hidden'
+                    },
+                    failed: {
+                        statusClass: 'error',
+                        iconClasses: 'fa-warning',
+                        shortMessage: 'Upload failed.',
+                        hiddenClass: ''
+                    },
+                    validationFailed: {
+                        statusClass: 'error',
+                        iconClasses: 'fa-warning',
+                        shortMessage: 'Validation failed.',
```

Comment:
One more user facing string

Reply:
Is there anything we can tell the user to try? What are the user's options if validation fails?